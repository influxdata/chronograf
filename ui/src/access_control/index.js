import RolesPageContainer from './containers/RolesPageContainer';
import RolePageContainer from './containers/RolePageContainer';
export {RolesPageContainer, RolePageContainer};
