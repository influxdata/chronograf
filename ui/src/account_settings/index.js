import React, {PropTypes} from 'react';
import {meShow, meUpdate} from 'shared/apis';
import FlashMessages from 'shared/components/FlashMessages';

const {func} = PropTypes;
const AccountSettingsPage = React.createClass({
  propTypes: {
    addFlashMessage: func.isRequired,
  },

  getInitialState() {
    return {};
  },

  componentDidMount() {
    meShow().then(({data: user}) => {
      this.setState({user});
    });
  },

  handleSubmit(e) {
    e.preventDefault();
    const email = this.email.value;
    const lastName = this.lastName.value;
    const firstName = this.firstName.value;
    const newUser = Object.assign({}, this.state.user, {lastName, firstName, email});
    meUpdate(newUser).then(() => {
      this.props.addFlashMessage({
        text: 'User updated successfully',
        type: 'success',
      });
    }).catch((err) => {
      this.props.addFlashMessage({
        text: err.toString(),
        type: 'error',
      });
    });
  },

  handleChangePassword(e) {
    e.preventDefault();
    const password = this.newPassword.value;
    const confirmation = this.newPasswordConfirmation.value;
    const newUser = Object.assign({}, this.state.user, {password, confirmation});
    meUpdate(newUser).then(() => {
      this.props.addFlashMessage({
        text: 'Password updated successfully',
        type: 'success',
      });

      // reset the form fields
      this.newPassword.value = '';
      this.newPasswordConfirmation.value = '';
    }).catch(() => {
      this.props.addFlashMessage({
        text: 'Passwords did not match',
        type: 'error',
      });
    });
  },

  render() {
    const {user} = this.state;
    if (!user) {
      return null; // or something else, idk
    }

    return (
      <div>
        <div className="enterprise-header">
          <div className="enterprise-header__container">
            <div className="enterprise-header__left">
              <h1>Account Settings</h1>
            </div>
          </div>
        </div>

        <div className="container-fluid">
          <div className="row">
            <div className="col-xs-12">
              <div className="panel panel-minimal">
                <div className="panel-heading">
                  <h2 className="panel-title">Personal Info</h2>
                </div>
                <div className="panel-body">
                  <form onSubmit={this.handleSubmit}>
                    <input type="hidden" />
                    <div className="form-group col-sm-4">
                      <label htmlFor="account-first-name">First Name</label>
                      <input ref={(ref) => this.firstName = ref} id="account-first-name" type="text" className="form-control" defaultValue={user.first_name} />
                    </div>
                    <div className="form-group col-sm-4">
                      <label htmlFor="account-last-name">Last Name</label>
                      <input ref={(ref) => this.lastName = ref} id="account-last-name" type="text" className="form-control" defaultValue={user.last_name} />
                    </div>
                    <div className="form-group col-sm-4">
                      <label htmlFor="account-email">Email Address</label>
                      <input ref={(ref) => this.email = ref} id="account-email" type="text" className="form-control" defaultValue={user.email} placeholder="Enter an email address" name="email" />
                    </div>
                    <div className="form-group col-sm-12">
                      <input type="submit" className="btn btn-primary" value="Change Info" />
                    </div>
                  </form>
                </div>
              </div>
              <div className="panel panel-minimal">
                <div className="panel-heading">
                  <h2 className="panel-title">Password</h2>
                </div>
                <div className="panel-body">
                  <form onSubmit={this.handleChangePassword}>
                    <input type="hidden" />
                    <div className="form-group col-sm-6">
                      <label htmlFor="account-new-password">New Password</label>
                      <input ref={(ref) => this.newPassword = ref} id="account-new-password" type="password" className="form-control" name="password" />
                    </div>
                    <div className="form-group col-sm-6">
                      <label htmlFor="account-confirm-new-password">Confirm New Password</label>
                      <input ref={(ref) => this.newPasswordConfirmation = ref} id="account-confirm-new-password" type="password" className="form-control" name="confirmation" />
                    </div>
                    <div className="form-group col-sm-12">
                      <input type="submit" className="btn btn-primary" value="Change Password" />
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  },
});

export default FlashMessages(AccountSettingsPage);
