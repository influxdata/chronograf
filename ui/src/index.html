<!DOCTYPE html>
<html>

<head>
  <meta http-equiv="Content-type" content="text/html; charset=utf-8" />
  <title>Chronograf</title>
</head>

<body>
  <div id='react-root' data-basepath=""></div>
  <script src="./index.tsx"></script>
</body>

</html>
