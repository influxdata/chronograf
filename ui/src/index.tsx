import 'babel-polyfill'

import * as React from 'react'
import {render} from 'react-dom'
import {Provider} from 'react-redux'
import {Route} from 'react-router'
import {ConnectedRouter} from 'react-router-redux'
import createHistory from 'history/createBrowserHistory'

import configureStore from './store/configureStore'
import {loadLocalStorage} from './localStorage'

import {withApp} from './App'
import {Login, UserIsAuthenticated, UserIsNotAuthenticated} from './auth'
import CheckSources from './CheckSources'
import {StatusPage} from './status'
import {HostsPage, HostPage} from './hosts'
import DataExplorer from './data_explorer'
import {DashboardsPage, DashboardPage} from './dashboards'
import AlertsApp from './alerts'
import {
  KapacitorPage,
  KapacitorRulePage,
  KapacitorRulesPage,
  KapacitorTasksPage,
  TickscriptPage,
} from './kapacitor'
import {AdminPage} from './admin'
import {SourcePage, ManageSources} from './sources'
import NotFound from 'shared/components/NotFound'

import {getMe} from 'shared/apis'

import {disablePresentationMode} from 'shared/actions/app'
import {
  authRequested,
  authReceived,
  meRequested,
  meReceived,
  logoutLinkReceived,
} from 'shared/actions/auth'
import {linksReceived} from 'shared/actions/links'
import {errorThrown} from 'shared/actions/errors'

import './style/chronograf.scss'

import {HEARTBEAT_INTERVAL} from 'shared/constants'

const errorsQueue = []

const rootNode = document.getElementById('react-root')

// Older method used for pre-IE 11 compatibility
const basepath = rootNode.getAttribute('data-basepath') || ''
window.basepath = basepath

const history = createHistory({
  basename: basepath, // this is written in when available by the URL prefixer middleware
})

const store = configureStore(loadLocalStorage(errorsQueue), history)
const {dispatch} = store

history.listen(() => {
  dispatch(disablePresentationMode())
})

window.addEventListener('keyup', event => {
  const escapeKeyCode = 27
  // fallback for browsers that don't support event.key
  if (event.key === 'Escape' || event.keyCode === escapeKeyCode) {
    dispatch(disablePresentationMode())
  }
})

class Root extends React.Component {
  private checkAuth = async () => {
    dispatch(authRequested())
    dispatch(meRequested())
    try {
      await this.startHeartbeat({shouldDispatchResponse: true})
    } catch (error) {
      dispatch(errorThrown(error))
    }
  }

  private startHeartbeat = async ({shouldDispatchResponse}) => {
    try {
      // These non-me objects are added to every response by some AJAX trickery
      const {data: me, auth, logoutLink, external} = await getMe()
      if (shouldDispatchResponse) {
        dispatch(authReceived(auth))
        dispatch(meReceived(me))
        dispatch(logoutLinkReceived(logoutLink))
        dispatch(linksReceived({external}))
      }

      setTimeout(() => {
        if (store.getState().auth.me !== null) {
          this.startHeartbeat({shouldDispatchResponse: false})
        }
      }, HEARTBEAT_INTERVAL)
    } catch (error) {
      dispatch(errorThrown(error))
    }
  }

  private flushErrorsQueue = () => {
    if (errorsQueue.length) {
      errorsQueue.forEach(errorText => {
        dispatch(errorThrown({status: 0, auth: null}, errorText, 'warning'))
      })
    }
  }

  public componentDidCatch(error: Error, info: React.ErrorInfo) {
    console.error(error, info)
  }

  public componentWillMount() {
    this.flushErrorsQueue()
    this.checkAuth()
  }

  public render() {
    return (
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <div>
            <Route path="/" component={UserIsAuthenticated(CheckSources)} />
            <Route path="/login" component={UserIsNotAuthenticated(Login)} />
            <Route
              path="/sources/new"
              component={UserIsAuthenticated(SourcePage)}
            />
            <Route path="/sources/:sourceID">
              <div>
                <Route component={CheckSources} />
                <Route path="status" component={withApp(StatusPage)} />
                <Route path="hosts" component={withApp(HostsPage)} />
                <Route path="hosts/:hostID" component={withApp(HostPage)} />
                <Route
                  path="chronograf/data-explorer"
                  component={withApp(DataExplorer)}
                />
                <Route path="dashboards" component={withApp(DashboardsPage)} />
                <Route
                  path="dashboards/:dashboardID"
                  component={withApp(DashboardPage)}
                />
                <Route path="alerts" component={withApp(AlertsApp)} />
                <Route
                  path="alert-rules"
                  component={withApp(KapacitorRulesPage)}
                />
                <Route
                  path="alert-rules/:ruleID"
                  component={withApp(KapacitorRulePage)}
                />
                <Route
                  path="alert-rules/new"
                  component={withApp(KapacitorRulePage)}
                />
                <Route
                  path="tickscript/new"
                  component={withApp(TickscriptPage)}
                />
                <Route
                  path="tickscript/:ruleID"
                  component={withApp(TickscriptPage)}
                />
                <Route
                  path="kapacitors/new"
                  component={withApp(KapacitorPage)}
                />
                <Route
                  path="kapacitors/:id/edit"
                  component={withApp(KapacitorPage)}
                />
                <Route
                  path="kapacitor-tasks"
                  component={withApp(KapacitorTasksPage)}
                />
                <Route path="admin" component={withApp(AdminPage)} />
                <Route
                  path="manage-sources"
                  component={withApp(ManageSources)}
                />
                <Route
                  path="manage-sources/new"
                  component={withApp(SourcePage)}
                />
                <Route
                  path="manage-sources/:id/edit"
                  component={withApp(SourcePage)}
                />
              </div>
            </Route>
            <Route path="*" component={NotFound} />
          </div>
        </ConnectedRouter>
      </Provider>
    )
  }
}

if (rootNode) {
  render(<Root />, rootNode)
}
