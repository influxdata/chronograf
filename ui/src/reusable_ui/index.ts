// Import Components
import Button from './components/Button'
import Dropdown from './components/dropdowns/Dropdown'
import MultiSelectDropdown from './components/dropdowns/MultiSelectDropdown'
import Form from './components/form_layout/Form'
import Input, {InputType} from './components/inputs/Input'
import OverlayTechnology from './components/overlays/OverlayTechnology'
import OverlayContainer from './components/overlays/OverlayContainer'
import OverlayHeading from './components/overlays/OverlayHeading'
import OverlayBody from './components/overlays/OverlayBody'
import PageHeader from './components/page_layout/PageHeader'
import PageHeaderTitle from './components/page_layout/PageHeaderTitle'
import Panel, {PanelType} from './components/panel/Panel'
import Radio from './components/radio_buttons/RadioButtons'
import SlideToggle from './components/slide_toggle/SlideToggle'

// Import Types
import {
  ComponentColor,
  ComponentSize,
  ComponentStatus,
  DropdownMenuColors,
  DropdownChild,
  ButtonShape,
  Greys,
  IconFont,
  Columns,
} from './types'

// Fire de lazer
export {
  Button,
  Dropdown,
  MultiSelectDropdown,
  Form,
  Input,
  InputType,
  OverlayTechnology,
  OverlayContainer,
  OverlayHeading,
  OverlayBody,
  PageHeader,
  PageHeaderTitle,
  Panel,
  PanelType,
  Radio,
  SlideToggle,
  ComponentColor,
  ComponentSize,
  ComponentStatus,
  DropdownMenuColors,
  DropdownChild,
  ButtonShape,
  Greys,
  IconFont,
  Columns,
}
