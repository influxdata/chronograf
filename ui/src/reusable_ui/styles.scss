// Reusable UI Components
@import "components/Button/Button";
@import "components/dropdowns/Dropdown";
@import "components/dropdowns/DropdownButton";
@import "components/form_layout/Form";
@import "components/inputs/Input";
@import "components/overlays/Overlay";
@import "components/panel/Panel";
@import "components/slide_toggle/SlideToggle";
@import "components/wizard/WizardCheckbox";
@import "components/spinners/loadingSpinner";
