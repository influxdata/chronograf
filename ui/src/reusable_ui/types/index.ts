export enum ComponentColor {
  Default = 'default',
  Primary = 'primary',
  Secondary = 'secondary',
  Success = 'success',
  Warning = 'warning',
  Danger = 'danger',
  Alert = 'alert',
}

export enum ComponentSize {
  ExtraSmall = 'xs',
  Small = 'sm',
  Medium = 'md',
  Large = 'lg',
}

export enum ComponentStatus {
  Default = 'default',
  Loading = 'loading',
  Error = 'error',
  Valid = 'valid',
  Disabled = 'disabled',
}

export enum DropdownMenuColors {
  Amethyst = 'amethyst',
  Malachite = 'malachite',
  Sapphire = 'sapphire',
  Onyx = 'onyx',
}

export type DropdownChild =
  | Array<string | JSX.Element | Element>
  | string
  | JSX.Element
  | Element

export enum ButtonShape {
  Default = 'none',
  Square = 'square',
  StretchToFit = 'stretch',
}

export enum Greys {
  Obsidian = '#0f0e15',
  Raven = '#1c1c21',
  Kevlar = '#202028',
  Castle = '#292933',
  Onyx = '#31313d',
  Pepper = '#383846',
  Smoke = '#434453',
  Graphite = '#545667',
  Storm = '#676978',
  Mountain = '#757888',
  Wolf = '#8e91a1',
  Sidewalk = '#999dab',
  Forge = '#a4a8b6',
  Mist = '#bec2cc',
  Chromium = '#c6cad3',
  Platinum = '#d4d7dd',
  Pearl = '#e7e8eb',
  Whisper = '#eeeff2',
  Cloud = '#f6f6f8',
  Ghost = '#fafafc',
  White = '#ffffff',
}

export enum IconFont {
  AddCell = 'add-cell',
  AlertTriangle = 'alert-triangle',
  Alerts = 'alerts',
  Annotate = 'annotate',
  AnnotatePlus = 'annotate-plus',
  AuthZero = 'authzero',
  BarChart = 'bar-chart',
  Capacitor = 'capacitor2',
  CaretDown = 'caret-down',
  CaretLeft = 'caret-left',
  CaretRight = 'caret-right',
  CaretUp = 'caret-up',
  Chat = 'chat',
  Checkmark = 'checkmark',
  Circle = 'circle',
  CircleThick = 'circle-thick',
  Clock = 'clock',
  CogOutline = 'cog-outline',
  CogThick = 'cog-thick',
  Collapse = 'collapse',
  CrownOutline = 'crown-outline',
  CrownSolid = 'crown2',
  Cube = 'cube',
  Cubouniform = 'cubo-uniform',
  DashF = 'dash-f',
  DashH = 'dash-h',
  DashJ = 'dash-j',
  Disks = 'disks',
  Download = 'download',
  Duplicate = 'duplicate',
  ExpandA = 'expand-a',
  ExpandB = 'expand-b',
  Export = 'export',
  Eye = 'eye',
  EyeClosed = 'eye-closed',
  EyeOpen = 'eye-open',
  GitHub = 'github',
  Google = 'google',
  GraphLine = 'graphline2',
  Group = 'group',
  Heroku = 'heroku',
  HerokuSimple = '',
  Import = 'import',
  Link = 'link',
  OAuth = 'oauth',
  Octagon = 'octagon',
  Okta = 'okta',
  Pause = 'pause',
  Pencil = 'pencil',
  Play = 'play',
  Plus = 'plus',
  PlusSkinny = 'plus-skinny',
  Pulse = 'pulse-c',
  Refresh = 'refresh',
  Remove = 'remove',
  Search = 'search',
  Server = 'server2',
  Shuffle = 'shuffle',
  Square = 'square',
  TextBlock = 'text-block',
  Trash = 'trash',
  Triangle = 'triangle',
  User = 'user',
  UserAdd = 'user-add',
  UserOutline = 'user-outline',
  UserRemove = 'user-remove',
  Wood = 'wood',
  Wrench = 'wrench',
  Star = 'star',
  Stop = 'stop',
  Zap = 'zap',
}

export enum Columns {
  One = 1,
  Two = 2,
  Three = 3,
  Four = 4,
  Five = 5,
  Six = 6,
  Seven = 7,
  Eight = 8,
  Nine = 9,
  Ten = 10,
  Eleven = 11,
  Twelve = 12,
}
