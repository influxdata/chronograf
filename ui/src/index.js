import React from 'react'
import {render} from 'react-dom'
import {Provider} from 'react-redux'
import {Router, Route, useRouterHistory} from 'react-router'
import {createHistory} from 'history'
import {syncHistoryWithStore} from 'react-router-redux'
import {bindActionCreators} from 'redux'

import configureStore from 'src/store/configureStore'
import {loadLocalStorage} from 'src/localStorage'

import App from 'src/App'
import {
  Login,
  UserIsAuthenticated,
  UserIsNotAuthenticated,
  Purgatory,
} from 'src/auth'
import CheckSources from 'src/CheckSources'
import {StatusPage} from 'src/status'
import {HostsPage, HostPage} from 'src/hosts'
import DataExplorer from 'src/data_explorer'
import {DashboardsPage, DashboardPage} from 'src/dashboards'
import AlertsApp from 'src/alerts'
import {
  KapacitorPage,
  KapacitorRulePage,
  KapacitorRulesPage,
  KapacitorTasksPage,
  TickscriptPage,
} from 'src/kapacitor'
import {AdminChronografPage, AdminInfluxDBPage} from 'src/admin'
import {SourcePage, ManageSources} from 'src/sources'
import NotFound from 'shared/components/NotFound'

import {getMeAsync} from 'shared/actions/auth'

import {disablePresentationMode} from 'shared/actions/app'
import {errorThrown} from 'shared/actions/errors'

import 'src/style/chronograf.scss'

import {HEARTBEAT_INTERVAL} from 'shared/constants'

const errorsQueue = []

const rootNode = document.getElementById('react-root')

// Older method used for pre-IE 11 compatibility
const basepath = rootNode.getAttribute('data-basepath') || ''
window.basepath = basepath
const browserHistory = useRouterHistory(createHistory)({
  basename: basepath, // this is written in when available by the URL prefixer middleware
})

const store = configureStore(loadLocalStorage(errorsQueue), browserHistory)
const {dispatch} = store

browserHistory.listen(() => {
  dispatch(disablePresentationMode())
})

window.addEventListener('keyup', event => {
  const escapeKeyCode = 27
  // fallback for browsers that don't support event.key
  if (event.key === 'Escape' || event.keyCode === escapeKeyCode) {
    dispatch(disablePresentationMode())
  }
})

const history = syncHistoryWithStore(browserHistory, store)

const Root = React.createClass({
  componentWillMount() {
    this.flushErrorsQueue()
    this.checkAuth()
  },

  async checkAuth() {
    try {
      await this.performHeartbeat({shouldResetMe: true})
    } catch (error) {
      dispatch(errorThrown(error))
    }
  },

  getMe: bindActionCreators(getMeAsync, dispatch),

  async performHeartbeat({shouldResetMe = false} = {}) {
    await this.getMe({shouldResetMe})

    setTimeout(() => {
      if (store.getState().auth.me !== null) {
        this.performHeartbeat()
      }
    }, HEARTBEAT_INTERVAL)
  },

  flushErrorsQueue() {
    if (errorsQueue.length) {
      errorsQueue.forEach(errorText => {
        dispatch(errorThrown({status: 0, auth: null}, errorText, 'warning'))
      })
    }
  },

  render() {
    return (
      <Provider store={store}>
        <Router history={history}>
          <Route path="/" component={UserIsAuthenticated(CheckSources)} />
          <Route path="/login" component={UserIsNotAuthenticated(Login)} />
          <Route path="/purgatory" component={UserIsAuthenticated(Purgatory)} />
          <Route
            path="/sources/new"
            component={UserIsAuthenticated(SourcePage)}
          />
          <Route path="/sources/:sourceID" component={UserIsAuthenticated(App)}>
            <Route component={CheckSources}>
              <Route path="status" component={StatusPage} />
              <Route path="hosts" component={HostsPage} />
              <Route path="hosts/:hostID" component={HostPage} />
              <Route path="chronograf/data-explorer" component={DataExplorer} />
              <Route path="dashboards" component={DashboardsPage} />
              <Route path="dashboards/:dashboardID" component={DashboardPage} />
              <Route path="alerts" component={AlertsApp} />
              <Route path="alert-rules" component={KapacitorRulesPage} />
              <Route path="alert-rules/:ruleID" component={KapacitorRulePage} />
              <Route path="alert-rules/new" component={KapacitorRulePage} />
              <Route path="tickscript/new" component={TickscriptPage} />
              <Route path="tickscript/:ruleID" component={TickscriptPage} />
              <Route path="kapacitors/new" component={KapacitorPage} />
              <Route path="kapacitors/:id/edit" component={KapacitorPage} />
              <Route path="kapacitor-tasks" component={KapacitorTasksPage} />
              <Route path="admin-chronograf" component={AdminChronografPage} />
              <Route path="admin-influxdb" component={AdminInfluxDBPage} />
              <Route path="manage-sources" component={ManageSources} />
              <Route path="manage-sources/new" component={SourcePage} />
              <Route path="manage-sources/:id/edit" component={SourcePage} />
            </Route>
          </Route>
          <Route path="*" component={NotFound} />
        </Router>
      </Provider>
    )
  },
})

if (rootNode) {
  render(<Root />, rootNode)
}
