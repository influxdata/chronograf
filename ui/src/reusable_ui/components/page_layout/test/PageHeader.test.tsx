import React from 'react'
import {shallow} from 'enzyme'

import {Page} from 'src/reusable_ui'

describe('PageHeader', () => {
  let wrapper

  const wrapperSetup = (
    overrideProps = {},
    overrideLeft = (
      <Page.Header.Left>
        <Page.Title title="Yeehaw" />
      </Page.Header.Left>
    ),
    overrideCenter = (
      <Page.Header.Center>
        <p>Center!</p>
      </Page.Header.Center>
    ),
    overrideRight = <Page.Header.Right />
  ) => {
    const props = {
      fullWidth: false,
      inPresentationMode: false,
      ...overrideProps,
    }

    return shallow(
      <Page.Header {...props}>
        {overrideLeft}
        {overrideCenter}
        {overrideRight}
      </Page.Header>
    )
  }

  it('mounts without exploding', () => {
    wrapper = wrapperSetup()
    expect(wrapper).toHaveLength(1)
  })

  it('matches snapshot with minimal props', () => {
    wrapper = wrapperSetup()
    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot in presentation mode', () => {
    wrapper = wrapperSetup({inPresentationMode: true})
    expect(wrapper).toMatchSnapshot()
  })

  it('should contain matching error text if incorrect child types are passed in', done => {
    try {
      wrapper = wrapperSetup({}, <p />, <p />, <p />)
    } catch (error) {
      expect(error.toString()).toEqual(
        'Error: <Page.Header> expected children of type <Page.Header.Left>, <Page.Header.Center>, or <Page.Header.Right>'
      )
      done()
    }
  })

  it('should contain matching error text if too many of one header type exist', done => {
    try {
      wrapper = wrapperSetup(
        {},
        <Page.Header.Right />,
        <Page.Header.Right />,
        <Page.Header.Right />
      )
    } catch (error) {
      expect(error.toString()).toEqual(
        'Error: <Page.Header> expects at most 1 of each child type: <Page.Header.Left>, <Page.Header.Center>, or <Page.Header.Right>'
      )
      done()
    }
  })
})
