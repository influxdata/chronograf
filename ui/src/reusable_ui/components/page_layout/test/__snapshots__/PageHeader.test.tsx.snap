// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader matches snapshot in presentation mode 1`] = `""`;

exports[`PageHeader matches snapshot with minimal props 1`] = `
<div
  className="page-header"
>
  <div
    className="page-header--container"
  >
    <PageHeaderLeft
      key=".0"
      offsetPixels={100}
    >
      <PageTitle
        title="Yeehaw"
      />
    </PageHeaderLeft>
    <PageHeaderCenter
      key=".1"
      widthPixels={200}
    >
      <p>
        Center!
      </p>
    </PageHeaderCenter>
    <PageHeaderRight
      key=".2"
      offsetPixels={100}
      showSourceIndicator={false}
    />
  </div>
</div>
`;
